### PR TITLE
Deploy agent v1.2.18

### PR DESCRIPTION
**Pull request by ankilosaurus**
deploy agent changes to include az and asg in ping request

Ready to merge.

### CODE REVIEW

**Comment by ankilosaurus** [comment]
@jaymecox-pinterest

File: deploy-agent/deployd/client/client.py [hunk 1]
```diff
@@ -29,31 +29,64 @@
...
             log.info("Use facter to get host info")
-            if not self._hostname and self._config.get_facter_name_key():
-                self._hostname = utils.get_info_from_facter(self._config.get_facter_name_key())
+            name_key = self._config.get_facter_name_key()
+            ip_key = self._config.get_facter_ip_key()
+            id_key = self._config.get_facter_id_key()
+            group_key = self._config.get_facter_group_key()
+            az_key = self._config.get_facter_az_key()
+            keys_to_fetch = set()
+            # facter call is expensive so collect all keys to fetch first
```
Comment (yongwen) [inline deploy-agent/deployd/client/client.py:53]: How expensive it is? is it just reading from cache in file? If it is related to the facter cache file lock, can we only call the facter once during the first deployd get_info_from_facter and store these info in a deployd property file? assuming these info will never change after the host launched.

Comment (ankilosaurus) [reply deploy-agent/deployd/client/client.py:53]: It's not related to cache file. For hostname, hostip, hostid, group, we currently exec "facter -p" 4 times(one for each). Jayme mentioned we can call "facter -p" with list of keys. So, converting those 4 calls into single call. 

Do we want to further put that in file and never call facter again?

Comment (yongwen) [reply deploy-agent/deployd/client/client.py:53]: r+ now. we can iterate to improve later.

**Pull request merged**